src/plugin/file/kml/kmlparser.js

```diff
@@ -1432,7 +1432,9 @@ plugin.file.kml.KMLParser.prototype.parseSchema_ = function(el) {
   var schemas = os.xml.getChildrenByTagName(el, 'Schema');
   for (var i = 0, n = schemas.length; i < n; i++) {
     var name = schemas[i].name || schemas[i].getAttribute('name');
-    this.kmlThings_.push(name);
+    if ((name != null) && (name != '')) {
```

Comment:
`if (name)` is sufficient to test that the value is defined and not an empty string. If you also want to defend against whitespace-only strings, `goog.string.isEmptyOrWhitespace(goog.string.makeSafe(name))` is a good (but unfortunately verbose) way to do that.

Reply:
After reviewing our current schema support, it seems you should skip the schema entirely if it doesn't provide a name. We don't support `SchemaData schemaUrl="#someSchemaId"` right now, so there is nothing further to do. I would recommend simply doing:
```
if (!name) {
  continue;
}
```